### PR TITLE
Added descriptor get_info() test

### PR DESCRIPTION
**Pull request by sammyshj**
Added the descriptor get_info() test. The test presently fails and work needs to be done on the parse_keywords() and get_info() method for the test to pass. This will be reflected in the next commits.

### CODE REVIEW

**Comment by sammyshj** [comment]
@meejah In commit https://github.com/sammyshj/txtorcon/commit/f9d771fa3fa1bc24e39358e56e1d4b957bc751d4, I have added a basic parse_get_info() method.
This method now handles both single line and multiline responses. However, the the method is still in initial stage and hence, the tests fail.
Let me know what you think about this and whether we should move forward with this.
Thanks

**Comment by sammyshj** [comment]
@meejah When the https://github.com/meejah/txtorcon/blob/master/test/test_torcontrolprotocol.py#L430 test is run, should the response be `a\nb\nc`, or the current response mentioned that is `\na\nb\nc`?

**Comment by meejah** [comment]
I think there should be a leading newline like the test says... why?

**Comment by sammyshj** [comment]
Okay I need to make necessary changes to the test https://github.com/sammyshj/txtorcon/commit/711ed48acc2f60e4595452115d039479bcd1d7b9#diff-87729881434e634ea8b65a976a61f2efR608 then

**Comment by sammyshj** [comment]
@meejah In commit https://github.com/sammyshj/txtorcon/commit/95d8557b5dc72b19b513d29c89da8192e99d3a37, the parsing is now done by first splitting the lines based on the newline character and then forming the dict by looking if the key is present in the passed arguments.
Now only one test fails, https://github.com/meejah/txtorcon/blob/master/test/test_torcontrolprotocol.py#L578
This is because we pass the argument as `FOO` however when the parsing is done, it finds the key as `a` which is not in the passed arguments and hence it fails. It expects the response to be, `250-FOO=one`Let me know what should be done about this test.
Thanks

**Comment by meejah** [comment]
Yeah broken test; change it to ask for "a" or answer with "FOO", whichever. Thanks! I'll look at the rest in a little while...

**Comment by sammyshj** [comment]
@meejah Fixed the test. Now, all the tests pass on Travis CI. Let me know if the code looks good to be merged and I will go ahead and squash all the commits into one.

File: txtorcon/torcontrolprotocol.py [hunk 1]
```diff
@@ -356,11 +357,19 @@ def get_info(self, *args):
...
             into a dict, you can use get_info_raw instead.
         """
-        return self.get_info_raw(*args).addCallback(parse_keywords)
+        lines = yield self.get_info_raw(*args)
+        rtn = {}
+        key = None
+        for line in lines.split('\n'):
+            if line.split('=', 1)[0] in args:
+                key = line.split('=', 1)[0]
+                rtn[key] = line.split('=', 1)[1]
```
Comment (meejah) [inline txtorcon/torcontrolprotocol.py]: i think you want + '\n' here and reverse them below too?

File: txtorcon/torcontrolprotocol.py [hunk 2]
```diff
@@ -356,11 +357,19 @@ def get_info(self, *args):
...
+            else:
+                rtn[key] = rtn[key] + '\n' + line
+        defer.returnValue(rtn)
```
Comment (meejah) [inline txtorcon/torcontrolprotocol.py]: as-in rtn[key] + line + '\n' (with the above change too)?

File: txtorcon/torcontrolprotocol.py (hunk 1 above)
Comment (sammyshj) [inline txtorcon/torcontrolprotocol.py]: If we get a response such as,
```
250-a=
foo
one
```
The first ```line.split('=', 1)[1]``` contains a value ```''```. Then the loop takes the next line and adds it to the current value so, the value now becomes, ```'' + '\n' + 'foo'``` to give us ```'\nfoo'```. Hence, the final value we get is, ```'\nfoo\none'```.
Let me know if you think that this works.

Comment (meejah) [inline txtorcon/torcontrolprotocol.py]: shouldn't it have a trailing newline as well?

Comment (sammyshj) [inline txtorcon/torcontrolprotocol.py]: For a multiline response, we do not have a trailing newline in the end. We can see this from the test, https://github.com/meejah/txtorcon/blob/master/test/test_torcontrolprotocol.py#L430

Comment (meejah) [inline txtorcon/torcontrolprotocol.py]: okay, cool. then yes, looks good and please squash.

**Comment by sammyshj** [comment]
@meejah Squashed all the commits. The PR can be merged now :)

**Pull request merged**